src/org/sosy_lab/java_smt/example/NQueens.java

```diff
@@ -147,69 +147,114 @@ BooleanFormula[][] getSymbols() {
         }
         return symbols;
     }
-    /*
-     * getRules is the method used to add constraints that ensure that no two queens are in the same
-     * row, column, or diagonal.
+    /**
+
+     * This method generates a list of rules that represent the constraints for the N-Queens problem
+
+     * @param symbols: a 2D boolean array representing the placement of the queens on the board
+
+     * @param context: SolverContext instance
+
+     * @return List<BooleanFormula>: a list of boolean formulas that represent the constraints for
+     *  the problem
      */
     @Override
     public List<BooleanFormula> getRules(BooleanFormula[][] symbols, SolverContext context) {
         List<BooleanFormula> rules = new ArrayList<>();
         int n = symbols.length;
 
-        // At least one queen per row
-        for (BooleanFormula[] pSymbol : symbols) {
-            List<BooleanFormula> clause = new ArrayList<>(Arrays.asList(pSymbol).subList(0, n));
+        /*Rule 1: At least one queen per row,
+        or we can say make sure that there are N Queens on the board
+        *
```

Comment:
Please start a comment with a whitespace, like `/* Rule..`.
Please end a comment where the comment ends, not two lines below.